### PR TITLE
Define top level Permission for dockerhub-latest-chart.yml (latest chart to Dockerhub) work flow

### PR DESCRIPTION
**Pull request by aditya7302**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Part of #5048 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: .github/workflows/dockerhub-latest-chart.yml [hunk 1]
```diff
@@ -3,6 +3,12 @@ on:
   push:
     branches:
       - master
+
+permissions:
+  contents: read     # Needed to fetch the repository contents.
+  packages: write    # Needed to push the chart to DockerHub.
```
Comment (liangyuanpeng) [inline .github/workflows/dockerhub-latest-chart.yml:9]: It's working for push to the ghcr.io, and i think we done't need it due to we push to the dockerhub.

**Comment by aditya7302** [comment]
@liangyuanpeng I have removed the redundant permission.

File: .github/workflows/dockerhub-latest-chart.yml (hunk 1 above)
Comment (liangyuanpeng) [reply .github/workflows/dockerhub-latest-chart.yml:9]:
```diff
---  permissions:
---  contents: read     # Needed to fetch the repository contents.
---  packages: write    # Needed to push the chart to DockerHub.

+++ permissions: read-all
```

I think we can ack the permissions of read-all .

**Comment by zhzhuang-zju** [comment]
/lgtm 
ask @liangyuanpeng to take another look

**Comment by liangyuanpeng** [comment]
This PR have two commit, and https://github.com/karmada-io/karmada/pull/5082/commits/39a667ed5fa0199e115d023c50138ed45005d2fe is open at https://github.com/karmada-io/karmada/pull/5081, so it mistakenly caused PR  https://github.com/karmada-io/karmada/pull/5081 to be merged as well when this PR is merged.

But the `Files Changed` only one file: `.github/workflows/dockerhub-latest-chart.yml`, this is the reason i approve it.  somewhat strange.


PR https://github.com/karmada-io/karmada/pull/5081 still have some comment, @aditya7302  could you open a new PR for it? Thanks.

**Comment by aditya7302** [comment]
@liangyuanpeng I have opened a new PR #5126 for PR #5081

**Pull request merged**